# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/Semaphore.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/Semaphore.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/angle/checkout/src/libANGLE/Semaphore.h	Sat Sep 07 13:38:36 2019 +0000
@@ -0,0 +1,56 @@
+//
+// Copyright 2019 The ANGLE Project Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// Semaphore.h: Defines the gl::Semaphore class [EXT_external_objects]
+
+#ifndef LIBANGLE_SEMAPHORE_H_
+#define LIBANGLE_SEMAPHORE_H_
+
+#include <memory>
+
+#include "angle_gl.h"
+#include "common/PackedEnums.h"
+#include "common/angleutils.h"
+#include "libANGLE/Error.h"
+#include "libANGLE/RefCountObject.h"
+#include "libANGLE/angletypes.h"
+
+namespace rx
+{
+class GLImplFactory;
+class SemaphoreImpl;
+}  // namespace rx
+
+namespace gl
+{
+class Context;
+
+class Semaphore final : public RefCountObject
+{
+  public:
+    Semaphore(rx::GLImplFactory *factory, GLuint id);
+    ~Semaphore() override;
+
+    void onDestroy(const Context *context) override;
+
+    rx::SemaphoreImpl *getImplementation() const { return mImplementation.get(); }
+
+    angle::Result importFd(Context *context, HandleType handleType, GLint fd);
+
+    angle::Result wait(Context *context,
+                       const BufferBarrierVector &bufferBarriers,
+                       const TextureBarrierVector &textureBarriers);
+
+    angle::Result signal(Context *context,
+                         const BufferBarrierVector &bufferBarriers,
+                         const TextureBarrierVector &textureBarriers);
+
+  private:
+    std::unique_ptr<rx::SemaphoreImpl> mImplementation;
+};
+
+}  // namespace gl
+
+#endif  // LIBANGLE_SEMAPHORE_H_